src/basic/path-util.c

```diff
@@ -639,14 +639,31 @@ static int check_x_access(const char *path, int *ret_fd) {
         return 0;
 }
 
-int find_executable_full(const char *name, bool use_path_envvar, char **ret_filename, int *ret_fd) {
+int find_executable_full(const char *name, const char *root, bool use_path_envvar, char **ret_filename, int *ret_fd) {
         int last_error, r;
         const char *p = NULL;
 
         assert(name);
 
         if (is_path(name)) {
                 _cleanup_close_ int fd = -1;
+                _cleanup_free_ char *path_name = NULL;
+
+                /* Function chase_symlinks is invoked only when root is not NULL,
+                 * as using it regardless of root value would alter the behavior
+                 * of existing callers */
+                if (root) {
+                        r = chase_symlinks(name,
+                                           root,
+                                           CHASE_PREFIX_ROOT|CHASE_SAFE|CHASE_NOFOLLOW,
```

Comment:
Drop CHASE_SAFE here. That's only really necessary when we are about to create/touch some files, but we don't do this here. The stuff we operate on is offline and conceptually read-only, hence no need to be more catholic than the pope here.

Reply:
hmm, also, CHASE_NOFOLLOW sounds unnecessary?